pageserver/src/http/routes.rs

```diff
@@ -1757,27 +1759,36 @@ async fn timeline_detach_ancestor_handler(
 
         let timeline = tenant.get_timeline(timeline_id, true)?;
 
-        let (_guard, prepared) = timeline
+        let progress = timeline
```

Comment:
Should this request accept a timeout query param?  I'm not sure what happens if the client times-out and retries the request while the work below is still ongoing.

Reply:
I've specified it as 503 but there should be a test rather than a better explanation. Thanks for pointing this out.

Reply:
Found the synchronization code - feel free to resolve